updatecli/scripts/docker-ce-updates.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# This script uses apt to find the latest version of docker-ce available.
+set -eu -o pipefail
+
+# Upadting the list of latest updates available for installed packages on the system
+apt-get update >/dev/null
+
+# Retrieve from apt-cache the latest version of docker-ce available
+# Note: apt-cache policy will return a result like:
+# :# apt-cache policy docker-ce
+# docker-ce:
+#   Installed: (none)
+#   Candidate: (none)
+#   Version table:
+# We want to get the Candidate version (which is the latest available)
+# And we want it in a readable format
+# --
+# DEV MEMO: Line explanation:
+# 1. Retrieve information about docker-ce from apt
+# 2. Keep only the line about the Candidate version (latest available)
+# 3. Cut it so we only keep the version and remove title (version contains a :, hence keeping fields 2 and 3)
+# 4. Trimming the result (removing spaces before and after)
+# 5. Surrounding the result with simple quotes ('') as it'll be needed this way
+apt-cache policy docker-ce | grep 'Candidate' | cut -f2,3 -d':' | xargs | { read -r x ; echo "'${x}'"; }
```

Comment:
```suggestion
apt-cache policy docker-ce `# 1. Retrieve information about docker-ce from apt` \
  | grep 'Candidate' `# 2. Keep only the line about the Candidate version (latest available)` \
  | cut -f2,3 -d':' `# 3. Cut it so we only keep the version and remove title (version contains a :, hence keeping fields 2 and 3)` \
  | xargs `# 4. Trimming the result (removing spaces before and after)` \
  | { read -r x ; echo "'${x}'"; } # 5. Surrounding the result with simple quotes ('') as it'll be needed this way
```

Proposal (e.g. nitpicking, not blocking and subject to yes or no answer) to provide comment per-instruction. it's a pattern I've learned a few years ago to help streamlining the cognitive load when reading comment on a complex shell pipeline set of instructions. WDYT?

Reply:
That's a good idea but what is the meaning of the backticks here?

Reply:
@olblak the backticks allow to inline comments (adding comments outside a backslash are syntax error, and starting a comment will comment the next piped instruction, so backticks creates a subshell with only the comment so it won't spawn a child process and it's not overloading the reading)

Reply:
Nice! I didn't know about that, and I was actually sad I couldn't add inline comments after backslashes :) Thanks a lot!